Sming/sming.mk

```diff
@@ -0,0 +1,593 @@
+
+#==============================================================================
+#   path settings
+#------------------------------------------------------------------------------
+BUILD_BASE      = out/build
+FW_BASE         = out/firmware
+
+#   default values unless specified by the user
+MODULES         ?= app                  
+EXTRA_INCDIR    ?= include              
+SPIFF_FILES     ?= files
+
+#   SDK directories 
+SDK_BASE	    ?= $(ESP_HOME)/sdk
+SDK_TOOLS	    ?= $(SDK_BASE)/tools
+SDK_ROMS        := $(SDK_BASE)/bin
+SDK_LIBDIR      := $(SDK_BASE)/lib
+SDK_INCDIR      := $(SDK_BASE)/include
+
+XTENSA_TOOLS    := $(ESP_HOME)/xtensa-lx106-elf/bin
+
+USER_LIBDIR     := $(SMING_HOME)/compiler/lib/
+THIRD_PARTY_DIR := $(SMING_HOME)/third-party
+
+#==============================================================================
+#   tools
+#------------------------------------------------------------------------------
+ESPTOOL2        := esptool2
+SPIFFY          := $(SMING_HOME)/spiffy/spiffy
+
+CC              := $(XTENSA_TOOLS)/xtensa-lx106-elf-gcc
+CXX             := $(XTENSA_TOOLS)/xtensa-lx106-elf-g++
+AR              := $(XTENSA_TOOLS)/xtensa-lx106-elf-ar
+LD              := $(XTENSA_TOOLS)/xtensa-lx106-elf-gcc
+OBJCOPY         := $(XTENSA_TOOLS)/xtensa-lx106-elf-objcopy
+OBJDUMP         := $(XTENSA_TOOLS)/xtensa-lx106-elf-objdump
+STRIP           := $(XTENSA_TOOLS)/xtensa-lx106-elf-strip
+
+PERL            ?= perl
+XXD             := xxd
+
+ifeq ($(MANUAL_RESET),1)
+	RESET_ESP       := $(PAUSE5) "*** PLEASE RESET ESP8266 NOW ***"
+endif   
+
+export COMPILE  := gcc
+export PATH     := $(XTENSA_TOOLS):$(PATH)
+
+#==============================================================================
+#   makefile logging
+#------------------------------------------------------------------------------
+V ?= $(VERBOSE)
+ifeq ("$(V)","1")
+	Q       :=
+	vecho   := @true
+else
+	Q       := @
+	vecho   := @echo
+endif
+
+#==============================================================================
+#   intermediate files
+#------------------------------------------------------------------------------
+#   name for the target project
+TARGET          = app
+
+APP_AR          := $(BUILD_BASE)/$(TARGET)_app.a
+TARGET_OUT_0    := $(BUILD_BASE)/$(TARGET)_0.out
+TARGET_OUT_1    := $(BUILD_BASE)/$(TARGET)_1.out
+
+#==============================================================================
+#   wifi settings
+#------------------------------------------------------------------------------
+#   we will use global WiFi settings from Eclipse Environment Variables, if possible
+WIFI_SSID       ?= ""
+WIFI_PWD        ?= ""
+
+ifneq ($(WIFI_SSID), "")
+	DEFINES += WIFI_SSID WIFI_PWD
+endif
+
+#==============================================================================
+#   do flash address calculations
+#------------------------------------------------------------------------------
+hexcalc          = $(shell $(PERL) -e"printf '0x%06X', $(1)")
```

Comment:
It would be better to depend on python here instead of asking the users of Sming to install also perl on their machines.